v2/googlecloud-to-googlecloud/src/main/java/com/google/cloud/teleport/v2/utils/BigtableUtils.java

```diff
@@ -0,0 +1,305 @@
+          entry.getTimestamp(),
+          getByteBufferFromString(String.valueOf(entry.getTimestampTo()), charset)
+      ));
+    }
+
+    if (entry.getValue() != null) {
+      // value
+      cells.add(new com.google.cloud.teleport.bigtable.BigtableCell(
+          BigtableUtils.bigtableRowColumnFamilyName,
+          ChangelogColumns.TIMESTAMP_FROM.getColumnNameAsByteBuffer(),
+          entry.getTimestamp(),
+          getByteBufferFromString(String.valueOf(entry.getValue()), charset)
+      ));
+    }
+
+    return new BigtableRow(
+        createChangelogRowKey(entry.getCommitTimestamp(), workerId, counter, charset),
+        cells
+    );
+  }
+
+  private static ByteBuffer getByteBufferFromString(String s, Charset charset) {
+    return ByteBuffer.wrap(s.getBytes(charset));
+  }
+
+  private static ByteBuffer createChangelogRowKey(
+      Long commitTimestamp,
+      String workerId,
+      AtomicLong counter,
+      Charset charset) {
+    String rowKey = (commitTimestamp.toString()
+        + BigtableUtils.bigtableRowKeyDelimiter
+        + workerId
+        + BigtableUtils.bigtableRowKeyDelimiter
+        + counter.incrementAndGet());
+
+    return ByteBuffer.wrap(rowKey.getBytes(charset));
+  }
+
+  public static List<ChangelogEntry> getValidEntries(
+      ChangeStreamMutation mutation,
+      HashSet<String> ignoreColumns,
+      HashSet<String> ignoreColumnFamilies,
+      Charset charset
+  ) {
+    // filter first and then format
+    List<ChangelogEntry> validEntries = new ArrayList<>();
+    for (Entry entry : mutation.getEntries()) {
+      if (entry instanceof SetCell) {
+        SetCell setCell = (SetCell) entry;
+        String familyName = setCell.getFamilyName();
+        String qualifierName ;
+        qualifierName = setCell.getQualifier().toString(charset);
+        if (isValidEntry(familyName, qualifierName, ignoreColumns, ignoreColumnFamilies)) {
+          validEntries.add(createChangelogEntry(mutation, entry));
+        }
+      } else if (entry instanceof DeleteCells) {
+        DeleteCells deleteCells = (DeleteCells) entry;
+        String familyName = deleteCells.getFamilyName();
+        String qualifierName;
+        qualifierName = deleteCells.getQualifier().toString(charset);
+        if (isValidEntry(familyName, qualifierName, ignoreColumns, ignoreColumnFamilies)) {
+          validEntries.add(createChangelogEntry(mutation, entry));
+        }
+      } else if (entry instanceof DeleteFamily) {
+        DeleteFamily deleteFamily = (DeleteFamily) entry;
+        String familyName = deleteFamily.getFamilyName();
+        if (isValidEntry(familyName, null, ignoreColumns, ignoreColumnFamilies)) {
+          validEntries.add(createChangelogEntry(mutation, entry));
+        }
+      }
+    }
+    return validEntries;
+  }
+
+  public static Charset getCharset(String charset) {
+    if (charset == null) {
+      return StandardCharsets.UTF_8;
+    }
+
+    return StandardCharsets.UTF_8;
```

Comment:
I think this was meant to get Charset.forName(charset)

Reply:
This was used for testing and the return statements were changed multiple times. 
Function is not used anymore, I'm removing it.